flow/layers/image_filter_layer.cc

```diff
@@ -80,7 +80,12 @@ void ImageFilterLayer::Preroll(PrerollContext* context) {
   filter_->map_device_bounds(filter_in_bounds, SkMatrix::I(),
                              filter_out_bounds);
   child_bounds.set(filter_out_bounds);
-  child_bounds.offset(offset_);
+  if (context->raster_cache) {
+    child_bounds.offset(SkScalarRoundToScalar(offset_.x()),
+                        SkScalarRoundToScalar(offset_.y()));
+  } else {
+    child_bounds.offset(offset_);
+  }
```

Comment:
I would just offset by `offset_`. None of the Preroll methods take integralTransform issues into account and that is an ongoing discussion about how to deal with that.

Reply:
I did this because the raster cache bounds get rounded-out. So if the child is at 99.9, IntegralTransform will snap it to 100, but the raster cache will capture from 99. It doesn't cause a position mismatch, but it means you get 1 blank row and/or column raster cached. But if there is another effort to fix it in a more general sense, I will take it out here.

Reply:
The child's bounds are also floating point so rounding the offset is only half of the problem. We round the sum, not the translation.

Reply:
Isn't this just as likely to cause a rounding problem as if you didn't round the offset - depending on the bounds of the children?

Reply:
This round here doesn't affect the draw position at all, just the bounds of the raster cache. I added it to make my tests more clear, as well as save on raster-caching 1 blank row and/or column in the case child_bounds would round up. But if I'm misunderstanding or it's too confusing just to have it handled here instead of generally, it can be removed without affecting the actual graphical output.